draft-ietf-privacypass-architecture.md

```diff
@@ -425,7 +425,54 @@ their position to try and reduce the anonymity sets that Clients belong
 to (or, user segregation). For each case, we provide mitigations that
 the Privacy Pass ecosystem must implement to prevent these actions.
 
-## Metadata Privacy Implications
+## Correlation attacks
+
+One of the requirements of the issuance protocol as described in {{issuance-protocol}}
+is that it provide unconditional input secrecy. This means that an issuance
+and its corresponding redemption events are unlinkable using the contents of
+a token. However, there are other ways in which these two events could be linked
+together, which we refer to as correlation.
+
+Correlation can be problematic from a privacy perspective. To demonstrate why,
+consider the lifecycle of a cross-origin token. Clients may invoke the issuance
+protocol in one context, e.g., when visiting site foo.example, but redeem the
+token in another context, e.g., when visiting site bar.example. If an attacker
+can correlate these two events, then they know that visits to foo.example and
+bar.example are related. In contrast, per-origin token contexts are the same,
+so correlating issuance and redemption only allows an attacker to learn when
+issuance and redemption correspond to the same context.
+
+In general, each issuance and redemption event is associated with some Client
+and context. For example, an issuance event might represent Client C running
+the issuance protocol to produce a token for website foo.example at time T.
+Let cI and sI be the Client and context pair associated with an issuance event.
+Similarly, let cR and sR be the Client and context pair associated with a
+redemption event. We refer to context correlation as the case where cI = cR,
+and source correlation as the case where sI = sR.
+
+Context correlation is possible through a number of different ways. The following
+list covers some examples:
+
+- The timestamps of issuance and redemption events for interactive tokens may be
+  sufficiently unique that an attacker can link the two by their timestamp. However,
+  even if these timestamps are not close, a small number of issuance and redemption
+  events can let an attacker link two events. For example, if there is only ever
+  a single issuance and redemption event in a day, then an attacker can link based
+  based solely on the presence of the events.
+- The origin for an issuance and redemption event can be unique for a given Client,
+  allowing an attacker to link both events.
+
+Depending on the Client identifiers used, context correlation may not constitute
+a privacy violation. In particular, because tokens are not bound to a single Client
+and therefore transferrable, an attacker cannot distinguish the scenarios where
+the same Client ran the issuance protocol and redemption protocol, or where one
+Client ran the issuance protocol and another client ran the redemption protocol.
```

Comment:
How realistic is the handing-off tokens case? If it's not very common, it doesn't seem like a strong mitigation.

Reply:
This isn't meant to suggest it as a mitigation. It's merely meant to say that an origin cannot guarantee that the same token which went through issuance is the same token which redeemed a token, because they're transferrable. (What part of the text suggests this is a mitigation?)

Reply:
It doesn't directly say it's a mitigation, I was just inferring that.

Reply:
Ah, okay, is there any way we could clarify? (@bemasc zeroed in on this as well, so surely the text could be improved.)

Reply:
Do we need this paragraph?

Reply:
I had the same read as tfpauly -- that this suggests a mitigation, but you're right that it doesn't. If it's not a common use case, why use this example? And I'm then wondering if removing it entirely removes anything importnat -- it seems to me that it doesnt.

Reply:
I think this paragraph is important because it notes that linking issuance and redemption together may not constitute an actual attack. Token transfer is a fundamental property of this protocol. This text acknowledges that property.

Reply:
I still don't think this text is helpful.  Most digital identification systems have this transferability property.  Passwords, private keys, even hardware tokens are all transferable between people.  Transferability is trivial unless there's a reason to expect it will be unusually common.

This text seems to be arguing against the strawman position that correlation _always_ harms privacy.  In general, I don't think that there is any technical action that _always_ harms privacy.  The important question is whether there is some real use case where an otherwise-dangerous thing becomes safe.  I would prefer to see the focus there.

Reply:
> I still don't think this text is helpful. Most digital identification systems have this transferability property. Passwords, private keys, even hardware tokens are all transferable between people. Transferability is trivial unless there's a reason to expect it will be unusually common.
> 
> This text seems to be arguing against the strawman position that correlation always harms privacy. In general, I don't think that there is any technical action that always harms privacy. The important question is whether there is some real use case where an otherwise-dangerous thing becomes safe. I would prefer to see the focus there.

I'm trying to understand you but am failing to do so. Surely there are technical actions that always harm privacy. For example, never using HTTPS is harmful, revealing your IP address and other fingerprinting information to an origin is harmful, allowing third-party cookies to track you is harmful, etc. 

Is your objection to the phrasing "privacy violation"? If not, what is it?

Reply:
In an attempt to find a compromise, here's a proposed rephrasing of the text:

~~~
Context correlation between two events may not constitute a privacy violation if 
there is no source correlation between those two events. For example, if issuance
for a visit to foo.example was correlated to redemption of a visit to bar.example,
yet the source for these two events was uncorrelated, then one only learns that these
two events are related and might be associated with the same source (Client). In
contrast, if the source of these two events were correlated, then one learns that 
the two visits correspond to the same source (Client), thereby leaking browsing history.
~~~

@tfpauly, @janaiyengar, @bemasc, is this an improvement?

Reply:
One way to frame the problem is that this text describes a "scenario" involving multiple clients (an "acquiring client" and "redeeming client"), but this scenario is not contemplated in Section 3, which defines the scope of this design as involving only 4 communicating parties.  We shouldn't be performing security analysis based on scenarios that aren't in scope.

It seems like this section inverts the ordinary logic of security analysis.  Instead of asking "is there any strange situation where this could be dangerous?", it asks "is there any strange situation where this could be safe?".

> For example, never using HTTPS is harmful

Yes, and this text is like "Depending on the content used, insecure HTTPS may not be a privacy violation.  In particular, because the HTTP payload may already be encrypted using HPKE, ...".  This is true, but it's not helpful unless that is a relevant, reasonably common use case.

Reply:
> One way to frame the problem is that this text describes a "scenario" involving multiple clients (an "acquiring client" and "redeeming client"), but this scenario is not contemplated in Section 3, which defines the scope of this design as involving only 4 communicating parties. We shouldn't be performing security analysis based on scenarios that aren't in scope.

I very strongly disagree with this statement! The protocol is defined in terms of 4 parties, but security and privacy need to account for multiple instances of each. This is fairly common practice.